lib/browser/mqtt.ts

```diff
@@ -291,26 +301,12 @@ export class MqttClientConnection extends BufferedEventEmitter {
      */
     async connect() {
         return new Promise<boolean>((resolve, reject) => {
-            reject = this._reject(reject);
-
-            try {
-                this.connection.on('connect',
-                    (connack: { sessionPresent: boolean, rc: number }) => {
-                        resolve(connack.sessionPresent);
-                        this.on_online(connack.sessionPresent);
-                    }
-                );
-                this.connection.on('error',
-                    (error: string) => {
-                        reject(`Failed to connect: error=${error}`);
-                    }
-                );
-                this.connection.on('message', this.on_message);
-                this.connection.on('offline', this.on_offline);
-                this.connection.on('end', this.on_disconnected);
-            } catch (e) {
-                reject(e);
-            }
+            this.connection.once('connect', (connack: IConnackPacket) => {
+                resolve(connack.sessionPresent);
+            });
+            this.connection.once('error', (error: Error) => {
+                reject(new CrtError(error));
+            });
```

Comment:
just FYI, when things weren't working on my machine (and I never figured out 100% why), I was seeing a "close" event from this.connection instead of an "error" event
not sure if we need to tackle that right now, just saying...